### ISSUE TITLE
Release Candidate - Minor Issues

### ISSUE THREAD

**Issue by ossie-git** [OWASP/ASVS#613]
Had a very high-level look at the RC and Control 1 which I hadn't look at before (the updated version).  Some issues:

pg. 3: Level 1 - first steps, automated, or whole of portfolio view -> First steps, automated, ...

pg. 7: Similarly, the following web sites are most likely to be useful to users/adopters of this standard
- I think this was accidentally added as a header.  I'd suggest renaming the section to "Others" and then have the above as the first line of the section

pg. 7:

Add a page break before Preface

pg. 9: Level 1 - first steps, automated, or whole of portfolio view
- same comment as above

pg. 13

V1.3 Session Management Architectural Requirements
This is a placeholder for future architectural requirements.
- Is this supposed to be completed before the launch?

pg. 14

1.2.2
"Verify that communications between application components, including APIs, middleware and data layers, are authenticated with the least necessary privileges. (C3)" -> "Verify that communications between application components, including APIs, middleware and data layers, are authenticated.  Components should have the least necessary privileges needed. (C3)"

pg. 15

V1.5 Input and Output Pipeline Architectural Requirements
- Remove the word Pipeline.  Output Pipeline sounds strange.

v1.5.2
"Verify that architecturally, serialized objects are not sent to untrusted clients, either entirely, or with adequate blob encryption or integrity controls to prevent deserialization attacks, replay, and object injection." -> Wording could be improved.  Also, encryption + integrity controls don't always protect against replay attacks by design so I dropped it.  Here's an alternative: "Verify that serialization is not used when communicating with untrusted clients.  If this is not possible, ensure that adequate integrity controls (and possibly encryption if sensitive data is sent) are enforced to prevent deserialization attacks including object injection."

v1.5.3
"Verify that input validation is enforced on a trusted service layer" -> What is a trusted service layer? Alternative: "Verify that input validation is enforced server-side.  If input is passed to an interpreter, input validation should occur close to or by the interpreter for which it is intended."

V1.12 Secure Files Architectural Requirements -> Rename: V1.12 File Upload Architectural Requirements

V1.13 API Architectural Requirements
- is this supposed to be completed before launch?

V1.14 Configuration Architectural Requirements
- Some of the controls here make more sense if they are in their own section (such as 1.14.1, 1.14.2, etc.).  I'd suggest renaming this section to: "Other Architectural Requirements"

I don't think I'll have time to review the rest of the document before it is released but I think it looks fine from what I saw previously

**Comment by tghosth** [comment]
Items not addressed:
> This is a placeholder for future architectural requirements.
> * Is this supposed to be completed before the launch?

I am guessing not :)

**Comment by vanderaj** [comment]
I changed from "server-side" to "trusted service layers" to cope with the movement away from owned infrastructure, where the trusted service layer might actually be somewhere deemed "trusted" (think secure boot, serverless, external party). We have to move beyond 3 tier servers owned by a single corp. I'll add some text to add explantory color to this. 

I want to leave room and make sure we had spots for 4.1. I expect there to be architectural issues there but we simply ran out of time in 4.0 to come up with any decent ones. 

I think let's leave 1.14 alone for now - it's too late for that change today. 

@tghosth - considering my replies, once I have added some explanatory text, I am going to close this ticket. Is that ok?
